jetty-server/src/main/java/org/eclipse/jetty/server/ResponseWriter.java

```diff
@@ -452,16 +452,35 @@ public PrintWriter format(String format, Object... args)
     }
 
     @Override
-    public PrintWriter format(Locale l, String format, Object... args)
+    public PrintWriter format(Locale locale, String format, Object... args)
     { 
         try 
         {
+            
+            /* If the passed locale is null then 
+            use any locale set on the response as the default. */
+            if(locale == null) 
+                locale = _locale;
+            
             synchronized (lock) 
             {
                 isOpen();
-                if ((_formatter == null) || (_formatter.locale() != l))
-                    _formatter = new Formatter(this, l);
-                _formatter.format(l, format, args);
+
+                if(_formatter == null)
+                {
+                    _formatter = new Formatter(this, locale);
+                } 
+                else if(_formatter.locale()==null)
```

Comment:
Field `_formatter` is only assigned in this block, and never with a null locale, so this branch seems redundant.

Reply:
I saw a code coverage test for this and all branches were taken.... so it is possible.